app/src/main/java/com/iic/mokojin/AddPlayerActivity.java

```diff
@@ -0,0 +1,199 @@
+
+        private ParseQueryAdapter<Person> mAdapter;
+
+        public AddPlayerFragment() {
+            setHasOptionsMenu(true);
+        }
+
+        @InjectView(R.id.people_list_view) ListView mPeopleListView;
+        @InjectView(R.id.person_name_edittext) EditText mPersonNAmeEditText;
+
+        @Override
+        public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+            super.onCreateOptionsMenu(menu, inflater);
+            // Inflate the menu; this adds items to the action bar if it is present.
+            inflater.inflate(R.menu.menu_add_player, menu);
+        }
+
+        @Override
+        public boolean onOptionsItemSelected(MenuItem item) {
+            // Handle action bar item clicks here. The action bar will
+            // automatically handle clicks on the Home/Up button, so long
+            // as you specify a parent activity in AndroidManifest.xml.
+            int id = item.getItemId();
+
+            //noinspection SimplifiableIfStatement
+            if (id == R.id.action_done) {
+                done(null);
+                return true;
+            }
+
+            return super.onOptionsItemSelected(item);
+        }
+
+        // If there is a person selected in the listview - returns it (as a fulfilled promise)
+        // Otherwise - creates a new person, and returns a promise which is fulfilled once that
+        // person has been created
+        private Task<Person> getPersonToJoin(@Nullable Person selectedPerson) {
+            if (null == selectedPerson) {
+                return new CreatePersonOperation().run(mPersonNAmeEditText.getText().toString());
+            } else {
+                return Task.forResult(selectedPerson);
+            }
+        }
+
+        @SuppressWarnings("unused")
+        @OnItemClick(R.id.people_list_view)
+        void onListViewItemClick(int position) {
+            done(mAdapter.getItem(position));
+        }
+
+        // param is null if nothing has been selected
+        private void done(@Nullable Person selectedPerson) {
+            getPersonToJoin(selectedPerson).continueWithTask(new Continuation<Person, Task<Player>>() {
+                @Override
+                public Task<Player> then(Task<Person> task) throws Exception {
+                    if (task.isFaulted()) {
+                        Log.e(LOG_TAG, "Error creating person", task.getError());
+                        throw task.getError();
+                    }
+                    return new JoinQueueOperation().run(task.getResult());
+                }
+            }).continueWith(new Continuation<Player, Void>() {
+                @Override
+                public Void then(Task<Player> task) throws Exception {
+                    if (task.isFaulted()) {
+                        Log.e(LOG_TAG, "Error joining queue", task.getError());
+                        throw task.getError();
+                    }
+                    getActivity().finish();
```

Comment:
extract this to done method so we can perform more things here, like prefill the player on return and stuff like that